### DISCUSSION TITLE
How to use staticCall

### DISCUSSION THREAD

**Discussion by yy415393462** [wevm/wagmi#2543] [Q&A]
wagmi's readContract works fine, but sometimes I need to read the return value of a contract function using a static call without using the ethers api. Does wagmi provide an api for this

**Comment by codeninja819** [comment]


**Reply by Elvaleryn** [reply]
No you can not, person is refererring to ether's callStatic. In wagmi/viem you can mimic it with either [usePrepareContractWrite ](https://wagmi.sh/react/prepare-hooks/usePrepareContractWrite)  or viem's simulate contract. [simulateContract](https://viem.sh/docs/contract/simulateContract.html#:~:text=simulateContract%20%E2%80%8B,the%20state%20of%20the%20blockchain.). 

A key point to note here is that if you're attempting to retrieve the result of a function with limited access, for example only a specific account can call, you should provide the account parameter as an address that has access to that function, or it will result in an error.

**Reply by DobromirKirovLime** [reply]
Which will be the access list address and storage key...? 
```
accessList: [
      {
        address: ???,
        storageKeys: ['???']
      }
    ]
```
